File: medical-clinic-app-front/src/app/interfaces/appointment.ts
import { StringifyOptions } from "querystring";
import { Patient } from "./patient";


export class Appointment {
    
    cnp: string;
    firstName: string;
    lastName: string;
    emailUser: string;
    numberUser: string;
    dataA: string;
    hour: string;
    
    patient: Patient;

    constructor(){}
}

File: medical-clinic-app/src/main/java/com/medicalclinicapp/medicalclinicapp/models/Appointment.java
package com.medicalclinicapp.medicalclinicapp.models;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

@Entity
@Getter
@Setter
@Table(name = "appointment")
@RequiredArgsConstructor
public class Appointment implements Comparable<Appointment> {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String cnp;
    private String firstName;
    private String lastName;
    private String emailUser;
    private String numberUser;
    private String dataA;
    private String hour;

    @Override
    public int compareTo(Appointment o) {
        return this.getDataA().compareTo(o.getDataA());
    }
}


